imports/plugins/core/cart/server/no-meteor/mutations/updateCartItemsQuantity.js

```diff
@@ -81,5 +81,8 @@ export default async function updateCartItemsQuantity(context, input) {
     updatedBy: userId
   });
 
-  return { cart: updatedCart };
+  // Re-fetch cart with updated data
+  const updatedCartAfterAppEvents = await Cart.findOne({ _id: cartId });
```

Comment:
I guess the only real question here is what is the performance impact of constantly refetching the cart. Certainly that's going to significantly impact how many times we grab from that db.

Reply:
I see where that could be an issue... although this is only called when items are updated once they are in the cart, not when items are added or removed from that cart, those are other mutations. Not saying that definitely makes it OK, but I think the frequency of this mutation being hit is much less than any other cart actions.

That being said, anytime we run this mutation pre-my update here, we hit `appEvents` which then hits the DB a few times throughout the different events. Is hitting the DB inside an `appEvent` with an `update` or `findOneAndUpdate` any different than what I'm doing here?

Reply:
So if you update the cart above, and an external process needs to then update the cart with additional/new data, what ensures that your new fetch will run **after** that data is completed, especially when that external process may be hitting an external API. It may work locally with custom rates but with something that takes longer it might not?